### PR TITLE
344: Create tasks for text labels

### PR DESCRIPTION
**Pull request by olliestanley**
This adds backend support for text label tasks for initial prompts and replies. Also adds support for the tasks to the discord-bot and the text-frontend, but not the website

### CODE REVIEW

File: backend/oasst_backend/prompt_repository.py
Comment (yk) [inline backend/oasst_backend/prompt_repository.py:285]: nice catch :)

File: backend/oasst_backend/prompt_repository.py
Comment (yk) [inline backend/oasst_backend/prompt_repository.py:301]: shouldn't `conversation` be a member here?

File: backend/oasst_backend/prompt_repository.py
Comment (yk) [inline backend/oasst_backend/prompt_repository.py:304]: same

File: oasst-shared/oasst_shared/schemas/protocol.py
Comment (yk) [inline oasst-shared/oasst_shared/schemas/protocol.py:286]: this will add a `user` member to `TextLabels`, which has some effects on other parts and might break protocol, although I agree the change is necessary. Thus, a few other places need to be changed:
- the prompt repository -> store_text_labels, the user needs to be stored. this probably even requires a database schema migration to save the user (like in the other interactions)
- the LabelTextRequest endpoint (api/v1/text_labels) should be adjusted
- website and bot requests that depend on the text_labels endpoint should be adjusted

I realize this is a lot. if you feel like this might be a bit too much, just change this back to `BaseModel` and we'll take it from there

File: text-frontend/__main__.py
Comment (yk) [inline text-frontend/__main__.py:14]: eventually we should probably send the valid labels in the task to the client

**Review by yk — COMMENTED** [review]
Thank you very much, really nice work. I've left some comments

**Comment by andreaskoepf** [comment]
@olliestanley would it be possible for you to join our OA discord server and ping me? https://discord.gg/HFCPfugy

File: backend/oasst_backend/prompt_repository.py
Comment (andreaskoepf) [inline backend/oasst_backend/prompt_repository.py:285]: good catch!

File: text-frontend/__main__.py
Comment (andreaskoepf) [reply text-frontend/__main__.py:14]: That's a good idea. Could we maybe also add a couple of positive labels?

Comment (andreaskoepf) [reply text-frontend/__main__.py:14]: ok, I see we have at least ... "creative", "beautiful",  "informative",

Comment (olliestanley) [reply text-frontend/__main__.py:14]: I used the same list which was already present in the `TextLabels` enum in `protocol.py`. Suggest we move any changes to the labels list to a separate issue/PR to avoid this one getting too broad

Comment (olliestanley) [reply text-frontend/__main__.py]: With the latest changes the backend should be sending the list of valid labels as part of the task. I still need to test to verify this works

**Review by yk — APPROVED** [review]
LGTM now, thank you very much! maybe let's wait for andreas' final comments

**Comment by yk** [comment]
also, once we merge, please post this into the breaking-changes channel and tag the web-team

**Comment by olliestanley** [comment]
Message ID is now sent out with the task as requested by @andreaskoepf and I have tested using the `text-frontend` to successfully complete label tasks. Will wait for any final comments from Andreas before merge and can then notify the web team of the breaking change in the `text_labels` endpoint

File: backend/alembic/versions/2023_01_05_1745-20cd871f4ec7_added_user_to_textlabels.py
Comment (andreaskoepf) [inline backend/alembic/versions/2023_01_05_1745-20cd871f4ec7_added_user_to_textlabels.py:14]: we need to fix the alembic tree, if you touch this PR .. could you please change down_revision to the latest?

File: backend/oasst_backend/models/text_labels.py
Comment (andreaskoepf) [inline backend/oasst_backend/models/text_labels.py:19]: could it be `primary_key=True` here is a mistake?

File: backend/oasst_backend/prompt_repository.py
Comment (andreaskoepf) [inline backend/oasst_backend/prompt_repository.py:414]: if we require that the sent message id it always the read server message UUID we should change the type to `TextLabels.message_id` to UUID.

File: oasst-shared/oasst_shared/schemas/protocol.py
Comment (andreaskoepf) [reply oasst-shared/oasst_shared/schemas/protocol.py:290]: Could you please make a notification in the #breaking-changes for this?

File: oasst-shared/oasst_shared/schemas/protocol.py
Comment (andreaskoepf) [inline oasst-shared/oasst_shared/schemas/protocol.py:296]: as mentioned above I think `message_id` should get type UUID

**Review by andreaskoepf — CHANGES_REQUESTED** [review]
overall looks good .. added some minor comments.

File: backend/alembic/versions/2023_01_05_1745-20cd871f4ec7_added_user_to_textlabels.py
Comment (olliestanley) [reply backend/alembic/versions/2023_01_05_1745-20cd871f4ec7_added_user_to_textlabels.py]: This should be fixed now but feel free to double check it's correct as I'm not too familiar with alembic

File: backend/oasst_backend/models/text_labels.py
Comment (olliestanley) [reply backend/oasst_backend/models/text_labels.py]: Yep good spot, removed

File: backend/oasst_backend/prompt_repository.py
Comment (olliestanley) [reply backend/oasst_backend/prompt_repository.py]: Have changed this to UUID

File: oasst-shared/oasst_shared/schemas/protocol.py
Comment (olliestanley) [reply oasst-shared/oasst_shared/schemas/protocol.py:290]: Will do on merge

**Review by andreaskoepf — APPROVED** [review]
ok, lets merge this! :-)

**Pull request merged**
